bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/Thing.java

```diff
@@ -136,4 +153,31 @@
      * @return true if thing is linked, false otherwise.
      */
     public boolean isLinked();
+
+    /**
+     * Returns an immutable copy of the {@link Thing} properties.
+     * 
+     * @return an immutable copy of the {@link Thing} properties (not null)
+     */
+    Map<String, String> getProperties();
+
+    /**
+     * Sets the property value for the property identified by the given name.
+     * 
+     * @param name the name of the property to be set (must not be null or empty)
+     * @param value the value of the property
+     * 
+     * @return the previous value associated with the name, or null if there was no mapping for the name or the previous
+     *         value was null
+     */
+    String setProperty(String name, String value);
+
+    /**
+     * Unsets the property value for the property identified by the given name.
+     * 
+     * @param name the name of the property to be unset (must not be null or empty)
+     * 
+     * @return the previous value associated with the name, or null if there was no mapping for the name
+     */
+    String unsetProperty(String name);
```

Comment:
Could also be implemented with null, but nevertheless I would prefer the current implemented variant. Its a more explicit and I don´t think that an additional method is a problem.